### PR TITLE
Add optional `ts` into `AbstractPipeline.forecast`

### LINKED ISSUE #1069: Add optional `ts` into `AbstractPipeline.forecast`
### 🚀 Feature Request

Add optional parameter `ts` into `AbstractPipeline.forecast` with default value `None`. It will allow users to use new `TSDatasets` after loading the pipeline.

### Proposal

1. Add optional parameter `ts` into `AbstractPipeline.forecast` with default value `None`. It will allow users to use new `TSDatasets` after loading the pipeline.
2. Update children.
  1. Change the signature in all the children.
  2. Add logic for checking if `ts` is given:
    * If `ts` is given then use `ts`, ignore its `transforms` during making the forecasting.
    * If `ts` isn't given then use `self.ts`, if it isn' present raise the error.
3. In notebook with inference demo add example with making forecast on a new `ts` without passing it during `load`.

### Test cases

1. Make sure current tests pass.
2. For every pipeline check.
  * That it is possible to make `forecast` on other `ts`. For example, keep only fraction of segments for check.
  * That it is possible to load pipeline from disk without `ts` and make `forecast` by passing a new `ts`.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Mr-Geekman**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

## Proposed Changes
Look #1069.

## Closing issues
Closes #1069.

### CODE REVIEW

File: etna/models/base.py [hunk 1]
```diff
@@ -631,7 +631,9 @@ def forecast(self, ts: "TSDataset", prediction_size: int) -> "TSDataset":
         predictions = self.raw_predict(test_dataset)
         future_ts = ts.tsdataset_idx_slice(start_idx=self.encoder_length, end_idx=self.encoder_length + prediction_size)
         for (segment, feature_nm), value in predictions.items():
-            future_ts.df.loc[:, pd.IndexSlice[segment, feature_nm]] = value[:prediction_size, :]
+            # we don't want to change dtype after assignment, but there can happen cast to float32
+            dtype = future_ts.df.loc[:, pd.IndexSlice[segment, feature_nm]]
+            future_ts.df.loc[:, pd.IndexSlice[segment, feature_nm]] = value[:prediction_size, :].astype(dtype)
```
Comment (martins0n) [inline etna/models/base.py:636]: May be we should cast to float64 instead if we suppose it

**Comment by Mr-Geekman** [comment]
<p>We have non-deterministic order of segments during plot. Do we want to fix that?</p>

---

 <a href='https://app.reviewnb.com/tinkoff-ai/etna/pull/1071/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by Mr-Geekman** [comment]
<p>I think it is related to <code>set</code> of segments somehow. </p>

---

 <a href='https://app.reviewnb.com/tinkoff-ai/etna/pull/1071/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

File: etna/models/base.py (hunk 1 above)
Comment (Mr-Geekman) [reply etna/models/base.py:636]: I just think that using current dtype will be more safe in the future.

**Review by martins0n — APPROVED** [review]

**Pull request merged**